SonicWall/sonicwall-fw/ingest/parser.yml

```diff
@@ -0,0 +1,355 @@
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{parsed_event.message.start or parsed_event.message.startTime}}"
+        output_field: date
+
+  - name: parse_end_date
+    filter: "{{parsed_event.message.end != null or parsed_event.message.endTime != null}}"
+    external:
+      name: date.parse
+      properties:
+        input_field:  "{{parsed_event.message.end or parsed_event.message.endTime}}"
+        output_field: date
+
+  # Parse 'rt' (as a timestamp or as a date)
+  - name: parse_receipt_time
+    external:
+      name: date.parse
+      properties:
+        input_field: "{{parsed_event.message.rt}}"
+        output_field: date
+
+  - name: set_timestamp
+  - name: set_cef_header_fields
+  - name: set_cef_extension_fields
+  - name: set_cef_extension_fields_cef
+  - name: set_ecs_fields
+  - name: set_sonicwall_fields
+
+stages:
+  set_timestamp:
+
+    actions:
+      # Default timestamp is the start or startTime field
+      - set:
+          '@timestamp': "{{parse_start_date.date}}"
+
+      # Second choice is the receipt time (rc) field
+      - set:
+          '@timestamp': "{{parse_receipt_time.date}}"
+        filter: "{{parse_start_date == null and parse_receipt_time != null}}"
+
+      # Last choice is the end or endTime field
+      - set:
+          '@timestamp': "{{parse_end_date.date}}"
+        filter: "{{parse_start_date == null and parse_receipt_time == null and parse_end_date != null}}"
+
+  set_cef_header_fields:
+  # CEF:Version|Device Vendor|Device Product|Device Version|Device Event Class ID|Name|Severity|[Extension]
+    actions:
+
+      - set:
+          observer.vendor: '{{parsed_event.message.DeviceVendor}}'
+          observer.type: '{{parsed_event.message.DeviceProduct}}'
+          observer.version: '{{parsed_event.message.DeviceVersion}}'
+          rule.id: '{{parsed_event.message.DeviceEventClassID}}'
+
+      - set:
+          # Match only for agentSeverity of type int
+          event.severity: '{{parsed_event.message.Severity}}'
+        filter: "{{parsed_event.message.Severity | re_match('[1-9]|10')}}"
+
+
+  set_cef_extension_fields:
+    actions:
+    
+      - set:
+          event.action: '{{parsed_event.message.act}}'
+          # network.protocol: '{{parsed_event.message.app}}' is id
+          event.reason: '{{parsed_event.message.msg}}'
+          file.name: '{{parsed_event.message.fname}}'
+          file.size: '{{parsed_event.message.fsize}}'
+          source.ip: '{{parsed_event.message.src or parsed_event.message.sourceAddress}}'
+          source.user.name: '{{parsed_event.message.suser}}'
+          destination.domain: "{{parsed_event.message.get('destination DnsDomain')}}"
+          service.name: "{{parsed_event.message.get('destination ServiceName')}}"
+          destination.nat.ip: "{{parsed_event.message.get('destination Translated Address')}}"
+          destination.nat.port: "{{parsed_event.message.get('destination TranslatedPort')}}"
```

Comment:
Could you add and fill [`event.kind`](https://www.elastic.co/guide/en/ecs/8.4/ecs-allowed-values-event-kind.html), [`event.category`](https://www.elastic.co/guide/en/ecs/8.4/ecs-allowed-values-event-category.html) and [`event.type`](https://www.elastic.co/guide/en/ecs/8.4/ecs-allowed-values-event-type.html)?